### DISCUSSION TITLE
IMAP APPEND: return new UID (APPENDUID)

### DISCUSSION THREAD

**Discussion by bohwaz** [curl/curl#11971] [Ideas]
When storing a new message on the IMAP server, following the documentation here: https://curl.se/libcurl/c/imap-append.html

curl doesn't return the new message UID, returned by the APPENDUID message from the server (as per the RFC 4315: https://www.rfc-editor.org/rfc/rfc4315 ).

Right now the only way to get the message uid is to enable CURLOPT_VERBOSE and use a custom STDERR handler that will parse the raw IMAP result. This is not optimal.

It would be nice if curl would return the new message UID (from APPENDUID) using CURLOPT_RETURNTRANSFER, if it is available, or null if the server does not return APPENDUID.

**Comment by bagder** [comment]
> It would be nice if curl would return the new message UID (from APPENDUID) using CURLOPT_RETURNTRANSFER

`CURLOPT_RETURNTRANSFER` does not exist in libcurl. It is a PHP invention that returns the response body.

Getting meta-data in the response body seems like the wrong thing. It seems like something worthy of adding a new option to `curl_easy_getinfo()` for.

**Reply by bohwaz** [reply]
Ah yes, absolutely, I wasn't sure of the way to get this information.
